vault/provider.go

```diff
@@ -157,6 +157,12 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("VAULT_SKIP_VERIFY", false),
 				Description: "Set this to true only if the target Vault server is an insecure development instance.",
 			},
+			"tls_server_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("VAULT_TLS_SERVER_NAME", false),
```

Comment:
Since `tls_server_name`'s type is a string, the default value should probably be `""` rather than a boolean; also we can use the constant from the `api` package for the env variable name.

```suggestion
				DefaultFunc: schema.EnvDefaultFunc(api.EnvVaultTLSServerName, ""),
```

Reply:
@benashz Thanks for the fix and merge!